civicrm.php

```diff
@@ -89,14 +89,21 @@
 
 // store PATH to this plugin's settings file
 if (!defined('CIVICRM_SETTINGS_PATH')) {
-  define( 'CIVICRM_SETTINGS_PATH', CIVICRM_PLUGIN_DIR . 'civicrm.settings.php' );
+  $upload_dir    = wp_upload_dir();
+  $wp_civi_settings = $upload_dir['basedir'] . DIRECTORY_SEPARATOR . 'civicrm' . 'civicrm.settings.php' ;
+  define( 'CIVICRM_SETTINGS_PATH', $upload_dir['basedir'] . DIRECTORY_SEPARATOR . 'civicrm'. DIRECTORY_SEPARATOR . 'civicrm.settings.php' );
+}
+
+// store 4.6 and prior version PATH to this plugin's settings file
+if (!defined('CIVICRM_SETTINGS_PATH_DEPRECATED')) {
+  define( 'CIVICRM_SETTINGS_PATH_DEPRECATED', CIVICRM_PLUGIN_DIR . 'civicrm.settings.php' );
```

Comment:
I think there are parts of the core code which rely on `CIVICRM_SETTINGS_PATH` pointing at the real/live copy of `civicrm.settings.php`. (Ex: The upgrader sometimes needs to patch the file for a new version.) In this code, I think we get the wrong value for legacy sites. Would it be ok to first do the check to determine the path -- and then set the constant CIVICRM_SETTINGS_PATH?